blog/2023-5-30-what-is-the-actor-model-and-when-should-i-use-it/index.mdx

```diff
@@ -0,0 +1,104 @@
+
+## How does XState work with the actor model?
+
+With XState, we expose the ability to create instances of machines as actors! Looking at the example below, we can see that after defining our machine and its attributes, we only need to use `interpret()` to instantiate an actor (called `toggleActor`) and send it messages.
+
+```typescript
+import { createMachine, interpret } from 'xstate';
+
+// State machine definition
+const toggleMachine = createMachine({
+  id: 'toggle',
+  initial: 'inactive',
+  states: {
+    inactive: { on: { TOGGLE: 'active' } },
+    active: { on: { TOGGLE: 'inactive' } },
+  },
+});
+
+// Machine instance with internal state
+const toggleActor = interpret(toggleMachine);
+toggleActor.subscribe((state) => {
+  console.log(state.value);
+});
+toggleActor.start();
+// => logs 'inactive'
+
+toggleActor.send({ type: 'TOGGLE' });
+// => logs 'active'
+
+toggleActor.send({ type: 'TOGGLE' });
+// => logs 'inactive'
+```
+
+This actor has its own state and context, and it can be updated whenever it receives events. Of course, in order for the internal state of an actor to be updated, the event must be a legal transition defined with the machine.
+
+### Spawning Actors
+
+XState can also be used to spawn other actors and communicate with each other:
+
+```typescript
+import { createMachine, spawn } from 'xstate';
+import { todoMachine } from './todoMachine';
+
+const todosMachine = createMachine({
+  // ...
+  on: {
+    'NEW_TODO.ADD': {
+      actions: assign({
+        todos: ({ context, event }) => [
+          ...context.todos,
+          {
+            todo: event.todo,
+            // add a new todoMachine actor with a unique name
+            ref: spawn(todoMachine, `todo-${event.id}`),
+          },
+        ],
+      }),
+    },
+    // ...
+  },
+});
+```
+
+With the use of `spawn()` and `assign()`, we create a new actor instance when provided the machine logic and a unique identifier.
+By their nature, actions are ["fire and forget" effects](https://xstate.js.org/docs/guides/actions.html), meaning they are executed with no expectation of receiving events back to the actor. This makes sense for creating a new actor, but we may still want the parent actor to have a reference to its child, so we save that in its context using `assign()`. `spawn()` is the function called that actually creates the new actor. The parent can access this state easily by calling `getSnapshot()` on the reference to the child.
+For more detailed examples around working with actors in XState, like callback or promised-based actor spawning, sending updates, and communicating between actors, [check out our guides](https://xstate.js.org/docs/guides/actors.html).
```

Comment:
```suggestion

For more detailed examples around working with actors in XState, like callback or promised-based actor spawning, sending updates, and communicating between actors, [check out our docs on actors](actors.mdx).
```

1. Put a line between this paragraph and the text above to break the text up and make it read more as a conclusion.
2. Update the link to reference our new docs
3. Make the link text a little more descriptive